File: 2018-12-23/1157/1157.c
#include <stdio.h>
#include <string.h>

int main(){
	char words[1000002];
	int result[26] = {0,};
	scanf("%s", words);
	int max = 0, count =0;
	int index;
	int length = strlen(words);

	for(int i = 0; i < length; i++){
		if(words[i] > 96){
			words[i] = words[i]-32;
		}
		result[words[i]-'A']+=1;
		}

	for(int i = 0 ;i<26;i++){
		if(max < result[i]){
			max = result[i];
			count = 0;
			index = i;
		}else if(max == result[i]){
			count+=1;
		}
	}

	if(count > 0)
		printf("?");
	else 
		printf("%c", index + 'A');
}

File: 2018-12-23/1157/1157.py
from sys import stdin 

words = stdin.readline().upper()
result = [0 for i in range(0, 26)]

for i in range(0, 26):
	result[i] = words.count(chr(i+65))

maxNum = max(result)

if(result.count(maxNum) >1):
	print("?")
else:
	print(chr(result.index(maxNum)+65))		
